pkg/cloud/config.go

```diff
@@ -2,297 +2,126 @@ package cloud
+			logger.Warn("Secret config path already set", log.String("configPath", opts.SecretOptions.SecretConfigPath))
+			return nil
 		}
-	}
 
-	configPath := getConfigPath()
-	if err := os.Remove(configPath); err != nil {
-		if !errors.Is(err, os.ErrNotExist) {
-			return err
+		configPath, err := getConfigFromTrivyCloud(ctx, client, opts, accessToken, ConfigTypeSecret)
+		if err != nil {
+			return xerrors.Errorf("failed to get secret config: %w", err)
 		}
-	}
-
-	return nil
-}
-
-// initFirstLogin initializes the default scanning settings to turn them on
-// after this, the user can configure in the config using the config set/unset commands
-func (c *Config) initFirstLogin() error {
-	if c.Token == "" {
-		// this isn't a login save, without a token it can't login
-		return nil
-	}
-
-	var firstLogin bool
-	_, err := keyring.Get(ServiceName, TokenKey)
-	if err != nil {
-		if !errors.Is(err, keyring.ErrNotFound) {
-			return err
+		if configPath != "" {
+			opts.SecretOptions.SecretConfigPath = configPath
 		}
-		firstLogin = true
-	}
-
-	if firstLogin {
-		// if first login, turn on all scanning options
-		c.Server.Scanning.Enabled = true
-		c.Server.Scanning.UploadResults = true
-		c.Server.Scanning.MisconfigConfig = true
-		c.Server.Scanning.SecretConfig = true
 	}
 	return nil
 }
 
-// Load loads the Trivy Cloud config from the config file and the keychain
-// If the config file does not exist the default config is returned
-func Load() (*Config, error) {
+// getConfigFromTrivyCloud downloads a config from Trivy Cloud and saves it to a file
+// it returns the path to the config file if it was downloaded successfully, otherwise it returns an error
+func getConfigFromTrivyCloud(ctx context.Context, client *http.Client, opts *flag.Options, accessToken string, configType ConfigType) (string, error) {
 	logger := log.WithPrefix(log.PrefixCloud)
-	var config Config
-	configPath := getConfigPath()
-	yamlData, err := os.ReadFile(configPath)
-	if err != nil {
-		if !errors.Is(err, os.ErrNotExist) {
-			return nil, err
-		}
-		logger.Debug("No cloud config file found")
-		defaultCopy := *defaultConfig
-		return &defaultCopy, nil
-	}
-	if err := yaml.Unmarshal(yamlData, &config); err != nil {
-		return nil, err
+	configTypeStr := string(configType)
+	configDir := filepath.Join(fsutils.TrivyHomeDir(), "cloud", configTypeStr)
+	if err := os.MkdirAll(configDir, os.ModePerm); err != nil {
+		return "", xerrors.Errorf("failed to create cloud config directory: %w", err)
 	}
 
-	token, err := keyring.Get(ServiceName, TokenKey)
-	if err != nil {
-		if !errors.Is(err, keyring.ErrNotFound) {
-			return nil, err
-		}
-		logger.Debug("No token found in keychain")
-		config.Token = ""
-		return &config, nil
+	configFilename := filepath.Join(configDir, "config.yaml")
+	// Return cached config if it was updated within the last hour
+	if stat, err := os.Stat(configFilename); err == nil && stat.ModTime().After(time.Now().Add(configCacheTTL)) {
```

Comment:
correct me, if i an wrong or missing something.
I see two problems with this:
	1.	After one hour, we’ll download the file again.
	2. If the user finds an issue in the config on the server and updates the file there, they’ll need to remove the local file to download the new one from the server.

Maybe we could send the hash of the local file to the server?
Server will have the hash of the config file and will compare it.
If the hash is empty (config file doesn’t exist) or doesn’t match, the server will send a new config file.
If the config is still valid, there’s no need to send the file again (unwrap it, etc.).